### PR TITLE
[Quest API] Add ClearAccountFlag() and GetAccountFlags() to Perl/Lua

### PR DESCRIPTION
**Pull request by Kinglykrab**
# Perl
- Add `$client->ClearAccountFlag(flag)`.
- Add `$client->GetAccountFlags()`.

# Lua
- Add `client:ClearAccountFlag(flag)`.
- Add `client:GetAccountFlags()`.

# Notes
- Made use of repositories and cleaned up existing code.

### CODE REVIEW

File: zone/lua_client.cpp [hunk 1]
```diff
@@ -1509,16 +1509,36 @@ bool Lua_Client::HasSpellScribed(int spell_id) {
...
 	Lua_Safe_Call_String();
 	return self->GetAccountFlag(flag);
 }
 
+luabind::object Lua_Client::GetAccountFlags(lua_State* L) {
+	auto t = luabind::newtable(L);
+	if (d_) {
+		auto self = reinterpret_cast<NativeType*>(d_);
+		auto l = self->GetAccountFlags();
+		auto i = 1;
```
Comment (Akkadius) [inline zone/lua_client.cpp:1532]: It feels a bit silly to have a simple type be `auto` - let's please just use the types for simple types it's easier to read

File: zone/perl_client.cpp [hunk 2]
```diff
@@ -1512,6 +1517,21 @@ std::string Perl_Client_GetAccountFlag(Client* self, std::string flag) // @categ
 	return self->GetAccountFlag(flag);
 }
 
+perl::array Perl_Client_GetAccountFlags(Client* self)
+{
+	perl::array result;
+
+	const auto& l = self->GetAccountFlags();
+
+	result.reserve(l.size());
```
Comment (Akkadius) [inline zone/perl_client.cpp:1526]: I dig it

File: zone/client.cpp [hunk 3]
```diff
@@ -7561,36 +7562,57 @@ void Client::SendFactionMessage(int32 tmpvalue, int32 faction_id, int32 faction_
...
 
-	accountflags[flag] = val;
+	AccountFlagsRepository::ClearFlag(database, e);
 }
 
-std::string Client::GetAccountFlag(std::string flag)
+void Client::SetAccountFlag(const std::string& flag, const std::string& value)
 {
-	return(accountflags[flag]);
+	auto e = AccountFlagsRepository::NewEntity();
```
Comment (Akkadius) [inline zone/client.cpp:7589]: Beautiful

**Review by Akkadius — COMMENTED** [review]
LGTM. 

My biggest thing is testing here. Have we tested flags in game?

File: zone/lua_client.cpp (hunk 1 above)
Comment (Kinglykrab) [reply zone/lua_client.cpp:1532]: For just `i` here?

**Comment by Kinglykrab** [comment]
> LGTM.
> 
> My biggest thing is testing here. Have we tested flags in game?

Yes, I tested these ingame and the functionality remains the same.

**Review by Akkadius — APPROVED** [review]

**Pull request merged**